android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -143,6 +148,40 @@ class TranslationsMiddleware(
         )
     }
 
+    /**
+     * Removes the site from the list of never translate sites using [scope] and dispatches the result to the
+     * store via [TranslationsAction.SetNeverTranslateSitesAction] or else dispatches the failure
+     * [TranslationsAction.TranslateExceptionAction].
+     *
+     * @param context Context to use to dispatch to the store.
+     * @param tabId Tab ID associated with the request.
+     * @param origin A site origin URI that will have the specified never translate permission set.
+     */
+    private fun removeNeverTranslateSite(
+        context: MiddlewareContext<BrowserState, BrowserAction>,
+        tabId: String,
+        origin: String,
+    ) {
+        engine.setNeverTranslateSpecifiedSite(
+            origin = origin,
+            setting = false,
+            onSuccess = {
+                logger.info("Success requesting never translate sites.")
```

Comment:
Recommend for `onSuccess`:
```
    context.store.dispatch(
        TranslationsAction.OperationRequestedAction(
            tabId = tabId,
            operation = TranslationOperation.FETCH_PAGE_SETTINGS,
        ),
    )
```
I think that'll switch the state of the page settings to the right option too if the page was impacted. (Could potentially do this in the reducer too, but I think it might be hard to know if the page origin was impacted or not.) I'd recommend a success condition test too, if the harness is cooperative.

Reply:
Added `FETCH_PAGE_SETTINGS` to success condition, as well as a success test